### ISSUE TITLE
`Style/ItBlockParameter` auto-correction can be unsafe

### ISSUE THREAD

**Issue by colszowka** [rubocop/rubocop#14081]
## Expected behavior

The [Style/itBlockParameter](https://docs.rubocop.org/rubocop/cops_style.html#styleitblockparameter) cop is documented as safe for auto-correction, so auto-correcting affected code should not break things.

## Actual behavior

Rubocop auto-correction doesn't properly detect some difference between using numbered arguments. I haven't yet figured out what exactly makes it happen that the `_1` is a single argument while `it` becomes an array, and the Ruby 3.4 docs I browsed also didn't mention any edge cases to consider here, but I have this repro-sample:

## Steps to reproduce the problem

```ruby
require "bundler/inline"

gemfile do
  source "https://rubygems.org"

  gem "rspec-core", "3.13.3"
  gem "rspec-expectations", "3.13.3"
end

require "rspec/autorun"
require "rspec/expectations"

RSpec::Matchers.define :example do
  chain(:numbered) { pp _1 }
  chain(:it) { pp it }
end

RSpec.describe "demo" do
  it { is_expected.to example.numbered(1).it(2) }
end
```

When ran, this is the output:

```
1
[2]
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.75.2 (using Parser 3.3.7.4, Prism 1.4.0, rubocop-ast 1.44.0, analyzing as Ruby 3.4, running on ruby 3.4.2) [x86_64-linux]
```

---

I'm not sure what the best course of action here would be - ideally the auto-correction works in a safe manner so we can adapt it safely, but at least it should be declared as unsafe auto-correction I think.

**Comment by Earlopain** [comment]
I have no idea why this happens but I'm certain this is a ruby bug. rspec is too magical for me to further reduce this so I pretty much just reported it with the code you provided here https://bugs.ruby-lang.org/issues/21256

**Comment by koic** [comment]
This is an interesting issue rooted in the core of Prism. Since it shouldn't result in a problem at the language specification level, it shouldn't be marked as unsafe. Unfortunately, there's nothing RuboCop can do in this case. The resolution will likely be handled within Prism. cc @kddnewton

**Comment by colszowka** [comment]
Thanks for the quick responses - I wasn't sure if it's something in ruby itself, nice that this was now identified and pin-pointed to an actual difference between the new and old parsers, I'll keep an eye on the filed ruby bug report so we can eventually auto-correct all occurences once the difference has been addressed 👍
